syne_tune/blackbox_repository/blackbox_tabular.py

```diff
@@ -150,15 +150,44 @@ def hyperparameter_objectives_values(self):
         :return: X, y of shape (num_evals * num_seeds * num_fidelities, num_hps)
         and (num_evals * num_seeds * num_fidelities, num_objectives)
         """
+        objectives_evaluations = self.objectives_evaluations
+        hyperparameters = self.hyperparameters
+        if np.isnan(np.sum(objectives_evaluations)):
+            # Replace nan with previous value. Assumes that elapsed time is cumulative.
```

Comment:
OK, so the outcome of this would be report the same error value for all later resources with the same elapsed_time?
This may lead to issues with the scheduler.

Reply:
I'd favor the following. We pass through nan values to the simulator backend, but make sure that there is only one (or no) switch from not-nan to nan, and then nan until the end.

The simulator backend would react to this by creating a stop event just after the last not-nan report. So this a stop decided by the training script, not the scheduler.

But this would, IMHO, not lead to issues with our schedulers, and it seems the cleanest solution to me.

Reply:
I think the solution you are suggesting makes sense (passing a sequence of consecutive nans and interpreting as a stop in the backend). 

However, this would require significant changes in other parts which may need some discussion. If the code Martin have is functional, I would be keen to ship it and iterate on this, it would be easy to change the behavior once we are sure our backend/schedulers can properly reacts to this.

Reply:
Makes sense